### PR TITLE
Master 09828 ranking fix

### PR DESCRIPTION
**Pull request by Shnoulle**
- add a new QuestionAttribute maxDBanswers, not updatable if survey is activated
- fix max_answers according to this attribute
- Have only maxDBanswers in survey runtime, then other are not set
- Sub question by answer number OR attribute in createFieldMap
- fix specific to EM, max_answers set with min(max_answers,maxDBanswers)
- todo: set maxDBanswers when update and when import LSA : actually : done just before updating Answer parts

### CODE REVIEW

**Comment by Shnoulle** [comment]
OK : think have all except the DB update part (but fix it before deleting answers if i don't make errors : i retest to be sure).

File: application/controllers/admin/database.php [hunk 1]
```diff
@@ -149,30 +149,31 @@ function index($sa = null)
...
-            $__max_db_answers = QuestionAttribute::model()->findByAttributes(array(
-                'qid' => $iQuestionID,
-                'attribute' => '__max_db_answers'
-            ));
+            /* for already activated survey and rank question type : fix the maxDbAnswer before deleting answers */
+            /* @todo : add it to upgrage DB system, and see for the lsa */
+            if($sQuestionType=="R" && Survey::model()->findByPk($iSurveyID)->active=="Y")
+            {
+                $oQuestionAttributeMaxDBanswers = QuestionAttribute::model()->find(
+                    "qid = :qid AND attribute = 'maxDBanswers'",
```
Comment (olleharstedt) [inline application/controllers/admin/database.php]: You are mixing naming conventions. "Maximum answers" is "max_anwsers", so I'd expect "maxDBanswers" to be "max_db_answers".

**Comment by olleharstedt** [comment]
What does this PR do that my commit didn't?

File: application/controllers/admin/database.php (hunk 1 above)
Comment (Shnoulle) [inline application/controllers/admin/database.php]: Yes, you're right, send the commit after.

**Comment by Shnoulle** [comment]
Allow adding more answer than the answer number at start.

Or : have 100 answer to chose but only 3 columns in DB. I can send a little lss file but must be tested with this commit.

**Comment by Shnoulle** [comment]
Naming conventyion fixed , but wait : i think there are a bug in LS2.50 with min_answers and max_answers

In 2.06 : you can use EM in same page. I think now : it don't work. Let me make a lss file + a bug report and fix it here.

Exemple survey : 

* MAX: Numeric question : "How many fruit do you love" 
* RANK : Rank your {MAX} loved fruit in order

**Comment by Shnoulle** [comment]
OK : 2.06 is better for such error, but still some issue. I send bug report today, but completely different issue

**Comment by Shnoulle** [comment]
Upload a lsa done with this pull merged
https://framadrop.org/r/ePiquT9z9v#TZk3FuCgqmd3QvDHOLZ/iawgbJFouCGJSrBAmO3cq78=

**Pull request merged**